### PR TITLE
[DO NOT MERGE] fix(loop): replace obsolete --message flag with -p in gh copilot invocations (wait)

### LINKED ISSUE #874: loop uses obsolete gh copilot --message flag
## Bug

`loop` and related watch/execute paths invoke `gh copilot --message ...`, but the currently installed GitHub CLI Copilot integration on this machine accepts `-p` rather than `--message`.

This causes the loop to fail immediately even when the squad setup itself is healthy.

## Environment

- `@bradygaster/squad-cli`: `0.9.4-insider.1`
- OS: WSL2 Ubuntu / Linux
- Node: `v22.22.1`
- npm: `10.9.4`
- GitHub CLI: authenticated and working
- `gh copilot --help` shows prompt usage as `gh copilot -p "..."`

## Repro

1. Create a valid `loop.md` with `configured: true`
2. Run:

```bash
npm exec squad -- loop
```

## Actual

The first round starts and then fails with:

```text
error: unknown option '--message'
Try 'copilot --help' for more information.
```

The failing constructed command is effectively:

```bash
gh copilot --message "...loop prompt..."
```

## Expected

`loop` should invoke the installed `gh copilot` in a way that is accepted by the current CLI.

## Evidence

On this machine:

```bash
gh copilot --help
```

shows examples like:

```bash
gh copilot -p "Summarize this week's commits"
```

and does not document `--message`.

## Root Cause

The installed Squad CLI still hardcodes `--message` in multiple places, including:

- `dist/cli/commands/loop.js`
- `dist/cli/commands/watch/index.js`
- `dist/cli/commands/watch/capabilities/*.js`

Example from the installed code:

```js
const args = ['copilot', '--message', prompt];
```

## Local workaround

Replacing `--message` with `-p` in the installed CLI allowed the loop to get past the previous immediate failure and begin Round 1 successfully.

## Suggested fix

Update the CLI integration points to use the currently supported `gh copilot` prompt flag shape, or centralize the Copilot command construction behind a compatibility layer so this does not break again when GitHub CLI changes.

### PR DESCRIPTION
**Pull request by diberry**
## Summary

Replaces the obsolete \--message\ flag with \-p\ in all \gh copilot\ invocations. The current GitHub CLI Copilot extension uses \-p\ (for prompt), not \--message\, causing loop/watch/capabilities to fail immediately.

## Changes

- **New:** \packages/squad-cli/src/cli/commands/copilot-args.ts\ — centralized \uildCopilotArgs(prompt, options)\ helper. Single source of truth for the Copilot CLI prompt flag.
- **Updated:** 8 files that each had their own copy of the command builder:
  - \loop.ts\ — \uildLoopAgentCommand\ now delegates to \uildCopilotArgs\
  - \watch/index.ts\ — \uildAgentCommand\ now delegates to \uildCopilotArgs\
  - 6 watch capabilities (execute, decision-hygiene, monitor-email, monitor-teams, retro, wave-dispatch) — replaced inline \uildAgentCommand\ with import
- **Tests:** New \	est/cli/copilot-args.test.ts\ with 12 test cases. Updated \watch-execute.test.ts\ to expect \-p\.

## Why centralize?

When the CLI surface changes again (it's happened before), we fix one file instead of eight. The helper also handles edge cases (whitespace-only agentCmd/copilotFlags) that the duplicated code didn't.

Working as GNC (Node.js Runtime)

Closes #874

### CODE REVIEW

**Comment by diberry** [comment]
Is this introducing issues that 916 is meant to fix?

File: packages/squad-cli/src/cli/commands/copilot-args.ts [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+ * so the prompt flag (`-p`) lives in exactly one place.  When the upstream
+ * CLI surface changes again we fix it here and nowhere else.
+ */
+
+export interface CopilotSpawnOptions {
+  /** Fully override the agent command (e.g., `custom-agent --flag`). */
+  agentCmd?: string;
+  /** Extra flags appended after the prompt (e.g., `--model gpt-4`). */
+  copilotFlags?: string;
+}
```
[the remark this answers is not in the document]
Comment (diberry) [reply packages/squad-cli/src/cli/commands/copilot-args.ts:14]: Fixed — added "Ignored when agentCmd is set" to the `copilotFlags` JSDoc in `CopilotSpawnOptions`.

File: packages/squad-cli/src/cli/commands/loop.ts [hunk 2]
```diff
@@ -134,15 +135,7 @@ function buildLoopAgentCommand(
...
-  if (options.agentCmd) {
-    const parts = options.agentCmd.trim().split(/\s+/);
-    return { cmd: parts[0]!, args: [...parts.slice(1), '--message', prompt] };
-  }
-  const args = ['copilot', '--message', prompt];
-  if (options.copilotFlags) {
-    args.push(...options.copilotFlags.trim().split(/\s+/));
-  }
-  return { cmd: 'gh', args };
+  return buildCopilotArgs(prompt, options);
```
[the remark this answers is not in the document]
Comment (diberry) [reply packages/squad-cli/src/cli/commands/loop.ts:138]: Fixed — `agentCmd` is now normalized once at the top of `runLoop()` via `options.agentCmd?.trim() || undefined` and used consistently for both the preflight gate and command construction.

**Comment by diberry** [comment]
## 🔍 Squad Review — Kaylee (Engineering)

| # | Check | Status | Notes |
|---|-------|--------|-------|
| 1 | Changelog entry | ✅ | .changeset/fix-copilot-message-flag.md (CLI patch) |
| 2 | Squashed to 1 commit | ✅ | 1 commit |
| 3 | CI green | ❌ | 	est job **failed** — investigate and fix before merge |
| 4 | Copilot comments resolved | ✅ | 3/3 threads resolved with fixes |
| 5 | No .squad/ files | ✅ | Clean |
| 6 | No unrelated files | ✅ | 13 files — all scoped to copilot args centralization |
| 7 | Tests for changes | ✅ | 	est/cli/copilot-args.test.ts (12 tests), watch-execute.test.ts + watch-capabilities.test.ts updated |
| 8 | Not a duplicate/reversal | ✅ | Unique scope (closes #874) |

**Verdict:** ❌ 1 blocker to fix — CI 	est job is failing. Mergeable state is locked.

---
*Review by Squad AI team (Kaylee — Engineering) · requested by Dina Berry*

**Review by diberry — COMMENTED** [review]
required: Branch is behind dev — needs rebase before merge.

suggestion: No minimum gh copilot version documented. Users on older versions will hit a cryptic failure with -p. A note in CHANGELOG or README would protect them.

suggestion: loop.test.ts has zero assertions that buildLoopAgentCommand produces -p. The centralized copilot-args.test.ts covers the helper well, but an integration-level check in loop.test.ts would be good.

nit: buildLoopAgentCommand is now a 1-line wrapper — call site could import buildCopilotArgs directly.
